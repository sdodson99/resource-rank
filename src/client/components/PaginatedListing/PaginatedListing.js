import React from 'react';
import PropTypes from 'prop-types';
import styles from './PaginatedListing.module.css';
import Listing from '../Listing/Listing';
import Pagination from '../Pagination/Pagination';

const PaginatedListing = ({
  children,
  selectedPage,
  pageCount,
  onPageClick,
}) => (
  <div className={styles.PaginatedListing} data-testid="PaginatedListing">
    <Listing>{children}</Listing>

    <div className="mt-8 flex justify-center">
      <Pagination
        selectedPage={selectedPage}
        pageCount={pageCount}
        onPageClick={onPageClick}
      />
    </div>
  </div>
);

PaginatedListing.propTypes = {
  children: PropTypes.node,
  selectedPage: PropTypes.number,
  pageCount: PropTypes.number,
  onPageClick: PropTypes.func,
};

PaginatedListing.defaultProps = {};

export default PaginatedListing;
