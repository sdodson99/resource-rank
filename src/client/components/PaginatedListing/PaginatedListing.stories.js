import React from 'react';
import PaginatedListing from './PaginatedListing';

export default {
  component: PaginatedListing,
  title: 'Components/PaginatedListing',
};

const Template = (args) => <PaginatedListing {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: [
    <div key="1">Child 1</div>,
    <div key="2">Child 2</div>,
    <div key="3">Child 3</div>,
  ],
};

export const PageCount = Template.bind({});
PageCount.args = {
  ...Default.args,
  pageCount: 5,
};

export const SelectedPage = Template.bind({});
SelectedPage.args = {
  ...Default.args,
  ...PageCount.args,
  selectedPage: 3,
};
