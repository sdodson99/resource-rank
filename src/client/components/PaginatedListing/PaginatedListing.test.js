import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { createRenderer } from 'react-test-renderer/shallow';
import PaginatedListing from './PaginatedListing';

describe('<PaginatedListing />', () => {
  let props;

  beforeEach(() => {
    props = {
      children: [<div key="1">Child 1</div>],
      pageCount: 3,
      selectedPage: 2,
    };
  });

  it('should mount', () => {
    render(<PaginatedListing {...props} />);

    const paginatedListing = screen.getByTestId('PaginatedListing');

    expect(paginatedListing).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const component = createRenderer().render(<PaginatedListing {...props} />);

    expect(component).toMatchSnapshot();
  });
});
