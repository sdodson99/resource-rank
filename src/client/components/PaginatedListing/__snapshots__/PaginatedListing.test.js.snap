// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginatedListing /> should render correctly 1`] = `
<div
  className="PaginatedListing"
  data-testid="PaginatedListing"
>
  <Listing>
    <div>
      Child 1
    </div>
  </Listing>
  <div
    className="mt-8 flex justify-center"
  >
    <Pagination
      pageCount={3}
      selectedPage={2}
      selectedPageGapMinRange={3}
      selectedPageRange={2}
    />
  </div>
</div>
`;
